org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/authentication/TrivialAccessController.java

```diff
@@ -254,4 +267,30 @@ public TrivialAuthConfig withLoginPageInstalled(final boolean loginPageInstalled
       return this;
     }
   }
+
+  /**
+   * Verify if the request is a browser /unload/ request. Unload requests are handled by {@code UnloadRequestHandler}.
+   */
+  protected boolean isUnloadRequest(final HttpServletRequest request) {
+    return UNLOAD_PATH_PATTERN.matcher(getTarget(request)).matches();
+  }
+
+  protected boolean isAuthenticatedRequest(final HttpServletRequest request) {
```

Comment:
This method could be changed to e.g. getAuthenticationStatus returning e.g. {principal, status: "validSubject" or "none" or "principalValid" or "principalInvalid" as enum} 
Then this method could be used in handleRequest() to 
1. detect if authenticated and cancel unload
2. continue chain if subject is valid
3. abort if principal is invalid
4. continueChainAsSubject if principal is valid
Currently the logic is very hard to read and duplicated.